src/icrar/leap-accelerate/cuda/device_tensor.h

```diff
@@ -55,7 +56,8 @@ namespace cuda
         , m_sizeDim1(sizeDim1)
         , m_sizeDim2(sizeDim2)
         {
-            size_t byteSize = sizeDim0 * sizeDim1 * sizeDim2 * sizeof(T);
+            size_t byteSize = GetByteSize();
+            std::cout << "tensor3:" << sizeDim0 << "x" << sizeDim1 << "x" << sizeDim2 << std::endl;
```

Comment:
Probably use logging instead? But I'm not sure what's the context this is executing on (host/device)

Reply:
I think I put this once I started seeing some memory related errors that I later found to be related to allocating ~8x more memory than required but is now resolved for cuda implementation. I should probably terminate on cuda errors now or put a guard here that checks the available cuda memory and throw if the tensor is going to exceed the available memory.

This class is a C++ object oriented utility for the host to handle unsafe cuda memory buffers as the Cuda api is C style and uses void** with cudaMalloc(). There are other other third party libraries that do this like https://github.com/CNugteren/CLCudaAPI but APIs like these tend to involve extra files for setting up kernels that can be invoked. I suspect in future the best interface to for would be combining SYCL and CUDA that also allows for single source compilation.